### PR TITLE
Remove Map usage from the server and mission browser.

### PR DESCRIPTION
**Pull request by pchote**
This adds support for loading map rules from a `MapPreview`, and removes the last uses of `Map` from places that don't actually want a full map object.  This makes the mission browser a bit smoother (the video button and description are now available instantly), and, once support is added to the resource center, give servers access to all the maps available on the resource center without having to manually sync.

This also checks off another major point in #10717, and means that we can now streamline Map initialization by dropping all the `Lazy`ness.

### CODE REVIEW

**Comment by pchote** [comment]
Dependencies have been merged.

File: OpenRA.Game/GameRules/RulesetCache.cs [hunk 1]
```diff
@@ -60,38 +62,48 @@ public Ruleset Load(IReadOnlyFileSystem fileSystem, Map map = null)
...
 		}
 
+		/// <summary>
+		/// Cache and return the Ruleset for a given map.
+		/// If a map isn't specified then return the default mod Ruleset.
+		/// </summary>
+		public Ruleset Load(IReadOnlyFileSystem fileSystem, Map map = null)
+		{
+			return Load(fileSystem,
+				map != null ? map.RuleDefinitions : NoMapRules,
```
Comment (abcdefg30) [inline OpenRA.Game/GameRules/RulesetCache.cs]: Couldn't we do something like
`return map == null ? Load(fileSystem, NoMapRules, NoMapRules,...) : Load(fileSystem, map.RuleDefinitions, map.WeaponDefinitions, ...);` here?

**Comment by pchote** [comment]
Fixed.  Now depends on #10902.

**Comment by pchote** [comment]
Rebased.

**Comment by obrakmann** [comment]
looks fine, :+1:

**Comment by abcdefg30** [comment]
Needs a rebase now.

**Comment by pchote** [comment]
Rebased.

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark:

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/74f2f112d6f4aae3caebb606a6a66073372e9c08...ae1f96cf882eb2602d1b5375ae44d151a19e6a1d)

**Comment by abcdefg30** [comment]
[Extra changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/44224062fccc88ca7f492a882f902d745bcf235c...d8f2d77b2a5f55ed38979beb1a30b54d1a77665e) as requested on IRC.

**Pull request merged**